### PR TITLE
ENH: added rsplit to StringMethods

### PR DESCRIPTION
**Pull request by mortada**
as a part of https://github.com/pydata/pandas/issues/9111 

cc @sinhrks

### CODE REVIEW

File: pandas/tests/test_strings.py [hunk 1]
```diff
@@ -1234,12 +1235,75 @@ def test_split(self):
...
 
+        # regex split
+        values = Series([u('a,b_c'), u('c_d,e'), NA, u('f,g,h')])
+        result = values.str.split('[,_]')
+        exp = Series([[u('a'), u('b'), u('c')],
+                      [u('c'), u('d'), u('e')], NA,
+                      [u('f'), u('g'), u('h')]])
+        tm.assert_series_equal(result, exp)
+
+    def test_rsplit(self):
```
Comment (sinhrks) [inline pandas/tests/test_strings.py]: Can you add some cases using  ``maxsplit`` and ``expand=True``?

File: pandas/core/strings.py [hunk 2]
```diff
@@ -734,6 +734,33 @@ def str_split(arr, pat=None, n=None):
     return res
 
 
+def str_rsplit(arr, pat=None, n=None):
+    """
+    Split each string in the Series/Index by the given delimiter
+    string, starting at the end of the string and working to the front.
+    Equivalent to :meth:`str.rsplit`.
+
```
Comment (sinhrks) [inline pandas/core/strings.py]: We're starting to use ``versionadded`` directive (#10215). Please add ``.. versionadded:: 0.16.2``, like:

- https://github.com/pydata/pandas/blob/master/pandas/core/generic.py#L2051

**Comment by sinhrks** [comment]
I think this is OK for 0.16.2 as a follow-up of string enh if there is no objections. Could you update:

- Release note
- ``text.rst`` and ``api.rst``

**Comment by mortada** [comment]
@sinhrks it's updated, please take a look

File: doc/source/whatsnew/v0.16.2.txt [hunk 3]
```diff
@@ -79,6 +79,8 @@ See the :ref:`documentation <basics.pipe>` for more. (:issue:`10129`)
 .. _magrittr: https://github.com/smbache/magrittr
 .. _R: http://www.r-project.org
 
+- Added `rsplit` to Index/Series StringMethods
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.2.txt]: add the PR number here as the issue number

File: pandas/tests/test_strings.py [hunk 4]
```diff
@@ -1234,12 +1235,75 @@ def test_split(self):
...
+        exp = Series([[u('a'), u('b'), u('c')],
+                      [u('c'), u('d'), u('e')], NA,
+                      [u('f'), u('g'), u('h')]])
+        tm.assert_series_equal(result, exp)
+
+        result = values.str.rsplit('_', expand=False)
+        tm.assert_series_equal(result, exp)
+
+        # regex split is not supported by rsplit
+        values = Series([u('a,b_c'), u('c_d,e'), NA, u('f,g,h')])
```
Comment (jreback) [inline pandas/tests/test_strings.py]: should this raise? (though not obvious how to detect this is a regex)

Comment (mortada) [inline pandas/tests/test_strings.py]: Yeah it's not clear how we would catch this - the user could in theory mean the literally string even if it looks like a regex

File: pandas/core/strings.py (hunk 2 above)
Comment (mortada) [inline pandas/core/strings.py]: it's added

File: pandas/tests/test_strings.py (hunk 1 above)
Comment (mortada) [inline pandas/tests/test_strings.py]: it's added

**Comment by jreback** [comment]
thanks!

**Pull request merged**